store/rootmultistore.go

```diff
@@ -295,6 +295,11 @@ func (rs *rootMultiStore) Query(req abci.RequestQuery) abci.ResponseQuery {
 		return res
 	}
 
+	if len(res.Proof) == 0 && len(res.Value) != 0 {
```

Comment:
For nil values, should we require a proof that the value is nil?

Reply:
Normally, if value is nil, we should return absence proof. But if the store is empty, both proof and value will be nil. In this case, we should return a multiStoreProof to demonstrage the store is empty.